dozer-sql/src/pipeline/expression/comparison.rs

```diff
@@ -690,10 +982,93 @@ pub fn evaluate_lt(
+        Field::String(ref left_v) | Field::Text(ref left_v) => match right_p {
             Field::String(ref right_v) => Ok(Field::Boolean(left_v < right_v)),
             Field::Text(ref right_v) => Ok(Field::Boolean(left_v < right_v)),
             Field::Null => Ok(Field::Null),
+            Field::UInt(right_v) => {
+                let left_val = u64::from_str(left_v).map_err(|_| {
+                    PipelineError::UnableToCast(format!("{}", right_v), "UInt".to_string())
+                })?;
+                Ok(Field::Boolean(left_val < right_v))
+            }
+            Field::U128(right_v) => {
+                let left_val = u128::from_str(left_v).map_err(|_| {
+                    PipelineError::UnableToCast(format!("{}", right_v), "U128".to_string())
+                })?;
+                Ok(Field::Boolean(left_val < right_v))
+            }
+            Field::Int(right_v) => {
+                let left_val = i64::from_str(left_v).map_err(|_| {
+                    PipelineError::UnableToCast(format!("{}", right_v), "Int".to_string())
+                })?;
+                Ok(Field::Boolean(left_val < right_v))
+            }
+            Field::I128(right_v) => {
+                let left_val = i128::from_str(left_v).map_err(|_| {
+                    PipelineError::UnableToCast(format!("{}", right_v), "I128".to_string())
+                })?;
+                Ok(Field::Boolean(left_val < right_v))
+            }
+            Field::Float(right_v) => {
+                let left_val = f64::from_str(left_v).map_err(|_| {
+                    PipelineError::UnableToCast(format!("{}", right_v), "Float".to_string())
+                })?;
+                Ok(Field::Boolean(left_val < right_v.0))
+            }
+            Field::Boolean(right_v) => {
+                let left_val = bool::from_str(left_v).map_err(|_| {
+                    PipelineError::UnableToCast(format!("{}", right_v), "Bool".to_string())
+                })?;
+                Ok(Field::Boolean(!left_val & right_v))
+            }
+            Field::Decimal(right_v) => {
+                let left_val = Decimal::from_str(left_v).map_err(|_| {
+                    PipelineError::UnableToCast(format!("{}", right_v), "Decimal".to_string())
+                })?;
+                Ok(Field::Boolean(left_val < right_v))
+            }
+            Field::Timestamp(right_v) => {
+                let left_val: DateTime<FixedOffset> = DateTime::<FixedOffset>::from_str(left_v)
+                    .map_err(|_| {
+                        PipelineError::UnableToCast(format!("{}", right_v), "Timestamp".to_string())
+                    })?;
+                Ok(Field::Boolean(left_val < right_v))
+            }
+            Field::Date(right_v) => {
+                let left_val = NaiveDate::from_str(left_v).map_err(|_| {
+                    PipelineError::UnableToCast(format!("{}", right_v), "Date".to_string())
+                })?;
```

Comment:
I think casting this automatically is a bit dangerous, we should keep the date format in consideration.
Probably an explicit cast is better in this case.

Reply:
fixed